goon/browserassets/js/browserOutput.js

```diff
@@ -176,20 +176,47 @@ function highlightTerms(el) {
 	if(regexHasError) return; //just stop right there ig the regex is gonna except
 	for (var i = 0; i < opts.highlightTerms.length; i++) { //Each highlight term
 		if(opts.highlightTerms[i]) {
+			var term = opts.highlightTerms[i];
 			if(!opts.highlightRegexEnable){
-				if(el.innerText.toString().toLowerCase().includes(opts.highlightTerms[i].toLowerCase())) //match normally
+				if(el.innerText.toString().toLowerCase().includes(term.toLowerCase())) //match normally
 				el.innerHTML = '<span style="background-color:'+opts.highlightColor+'">'+el.innerHTML+'</span>' //encloseincludes
 				continue;
 			}
 			var rexp;
 			try{
-				rexp = new RegExp(opts.highlightTerms[i],"gmi")
+				rexp = new RegExp(term,"gmi")
 			} catch(e){
-				el.innerHTML+='<br/><span style="boldwarning"> Your highlight regex - '+opts.highlightTerms[i]+' - is malformed. Thrown exception: '+e+'</span>'
+				el.innerHTML+='<br/><span style="boldwarning"> Your highlight regex - '+term+' - is malformed. Thrown exception: '+e+'</span>'
 				regexHasError = true;
 				return;
 			}
-			el.innerHTML = el.innerHTML.replace(rexp,"<span style=\"background-color:"+opts.highlightColor+"\">$0</span>") //i cant figure out a proper, non snowflakey way to let people select the group that gets highlighted
+			
+			// highlight using regex without destroying HTML
+			for (var j = 0; j < el.childNodes.length; j++) {
+				var child = el.childNodes[j];
+				if (child.nodeType != Node.TEXT_NODE) {
+					if (child.className != "highlighted")
+						highlightTerms(child);
+					continue;
+				}
+					
+				var tx = child.textContent;
+
+				var index = tx.toLowerCase().indexOf(term.toLowerCase());
+				if (index > -1) {
```

Comment:
This won't highlight multiple matches right? So if you highlight "ai" and you have the sentence "AI open the door or I'll kill myself AI. Law 1 AI". Will it match all 3 of those?

Reply:
Looks like it does, but it doesn't keep the case on the highlighted stuff:

![image](https://user-images.githubusercontent.com/1496804/84567835-a475fd00-ad7b-11ea-8420-8f66eaa3dade.png)

I'll push a fix for that shortly

Reply:
No this one does what I expect it to do. Except for the casing ofcourse. Thanks for testing!
I also now see how it goes right. It creates a new element for the words behind and thus those will also be checked in the DOM loop